### DISCUSSION TITLE
Guidence for Newcomers

### DISCUSSION THREAD

**Discussion by kzlsahin** [AvaloniaUI/Avalonia#16301] [Q&A]
Hi.

I am looking an entry point to contribute to the Avalonia project, as I am impressed with the project, I believe its potential and willing to be part of it. I've been reading your guides about contribution and examining issues.

I feel hesitant when I see 1.3k issues and a long history. It seems really tough to pick an appropriate issue for a newcomer who is unfamiliar with the ongoing process. It just makes me feel like I have to jump onto a moving train.

Would you offer some advice? Do we have guidance for newcomers? I would really like to be able to contribute by using my coding skills.

**Comment by timunie** [comment]
I started like this:
1. try to run the samples especially ControlCatalog and play around with it
2. See which issue is probably easy to solve and or an issue that is bugging you. 
3. Try to fix it in source. If possible, add a failing test first. 

4. Any help is welcome. Not only PRs but also community support ny answering open discussions or helping out other in the telegram channel. 

Welcome 🤗

**Reply by kzlsahin** [reply]
Great, thank you.

I am excited to begin.

**Comment by stevemonaco** [comment]
Docs are an easy to get into area that also needs a lot of work. At this point, Avalonia itself is a long-lived project. These are harder to contribute to without knowing a sizable amount about the code base, design, and general XAML ecosystem. Issues marked with help-wanted are typically less complex to implement and touch fewer things in the code base: https://github.com/AvaloniaUI/Avalonia/issues?q=is%3Aissue+is%3Aopen+label%3Ahelp-wanted

**Reply by kzlsahin** [reply]
Thank you very much for the hints.

Then I will focus on the `avalonia-docs` and the issues of `avalonia` marked with help-wanted.

I have a question, though. Are there any unwritten customs about picking an issue and starting to work on it, such as notifying others or assigning it to oneself?

**Reply by thevortexcloud** [reply]
There are no specific processes I know of beyond being the first to submit a PR. Although issues do sometimes get picked up by the core team _eventually_ (in some cases years later if it was blocked by some other feature). For complex features/things likely to break other things though it's best to ask or have a discussion first about what you want to do before you go write any code. As there may already be existing plans somewhere or your implementation may conflict with something else.

Anyway, as for the docs stuff as long as it's accurate and relevant to Avalonia, you probably wont run into much issue. The docs repo is here:

https://github.com/AvaloniaUI/avalonia-docs/

There is also currently a small list of docs that need to be written if you want to pick one up. I think _some_ of them have already been assigned though.

https://github.com/AvaloniaUI/avalonia-docs/issues/384

**Reply by MikeCodesDotNET** [reply]
The [contributing guide ](https://github.com/AvaloniaUI/Avalonia/blob/master/CONTRIBUTING.md) should be required reading if you're keen to get stuck in.

**Reply by kzlsahin** [reply]
> The [contributing guide ](https://github.com/AvaloniaUI/Avalonia/blob/master/CONTRIBUTING.md) should be required reading if you're keen to get stuck in. 

Thank you for the reminder.

I have already read it four times and took notes :))

Though, if I do something wrong, hope you pardon me for the begining.
